h5py/_hl/files.py

```diff
@@ -418,6 +442,10 @@ def __init__(self, name, mode='r', driver=None,
             raise ValueError(
                 "h5py was built without ROS3 support, can't use ros3 driver")
 
+        if locking is not None and hdf5_version < (1, 12, 1) and (
+                hdf5_version[:2] != (1, 10) or hdf5_version[2] < 7):
+            raise ValueError("HDF version >= 1.12.1 or 1.10.x >= 1.10.7 required for file locking.")
```

Comment:
```suggestion
            raise ValueError("HDF version >= 1.12.1 or 1.10.x >= 1.10.7 required for file locking options.")
```

Just to make it slightly clearer - older versions already did file locking, they just didn't have this way to control it.